ImageWrapper.cpp

```diff
@@ -0,0 +1,70 @@
+#include "lodepng.h"
+#include <vector>
+#include <iostream>
+#include "ImageWrapper.h"
+//#include "Log.h" idk why but this isnt working
+
+
+
+
+std::vector<unsigned char> ImageWrapper::decodeImageToRGBAVector(const char* filename) 
+{
+
+	std::vector<unsigned char> imagepixels;
+	 //the raw pixels, size: 8 bits, range: 0 -> 255
+	unsigned int width, height;
+
+	//decode
+	unsigned error = lodepng::decode(imagepixels, width, height, filename);
+
+	//if there's an error, display it
+	if (error) std::cout << "decoder error " << error << ": " << lodepng_error_text(error) << std::endl;
+
+	return imagepixels;
+}
+
+//R, G, B, A, R, G, B, A, R, G, B, A, R, G, B, A <- RGBAVector
+//increments by 4 through RGBAVector to fill out pixel struct, then adds to pixelVectorToReturn then returns it.
+//maybe some optimization here somehow through pointers/refs, i should definitely not pass
+//an entire copy of RGBAVector
+std::vector<pixel> ImageWrapper::convertRGBAVectorToPixelVector(std::vector<unsigned char>& RGBAVector)
+{
+	std::vector<pixel> pixelVectorToReturn;
+
+	for (int i = 0; i < RGBAVector.size(); i+=4) {
+		struct pixel p;
+		p.r = RGBAVector.at(i);
+		p.g = RGBAVector.at(i + 1);
+		p.b = RGBAVector.at(i + 2);
+		p.a = RGBAVector.at(i + 3);
+		pixelVectorToReturn.push_back(p);
+	}
+
+	return pixelVectorToReturn;
```

Comment:
This should fine, in older c++ it would copy the whole local vector you created here when you return it, but since C++11, std::vector has move-semantics, meaning the local vector you create here should be moved upon return.

Reply:
And I believe that you have this project set to use C++14 standards